frontend/src/views/SQRAllotment.vue

```diff
@@ -136,23 +138,33 @@ export default {
     },
     filter: {
       deep: true,
-      handler: function() {
+      handler: async function() {
         this.files = null;
         this.allotment.files = [];
 
         if (this.filter.list == null) return;
 
-        this.$http
-          .jsonp(process.env.VUE_APP_SCRIPT_URL, {
-            params: {
-              path: "sqr/files",
-              list: this.filter.list,
-              language: this.filter.language
-            }
-          })
-          .then(response => {
-            this.files = response.body;
-          });
+        const fetchData = await fb
+          .database()
+          .ref(`sqr/files/${this.filter.list}`)
+          .once("value");
+        const files = fetchData.val();
```

Comment:
Here you actually pull all the records and filter in the browser. Isn't it possible to [filter server-side](https://firebase.google.com/docs/database/admin/retrieve-data#section-queries). Maybe for this sake we'd need to create [indexes in db](https://firebase.google.com/docs/database/security/#section-defining-indexes).

Reply:
I am having trouble figuring out the best way to filter server-side since I need to filter by two fields but firebase only allows one field. Do you have any suggestions?

Reply:
Seems you're right. According to [this answer](https://stackoverflow.com/a/26701282) I propose that we rethink the data structure and build our custom indexes in the DB. Really, NoSQL is different, and the data cannot be normalized as in SQL. So we'd have to push data into several nodes and/or move the data between nodes on some changes (allotments etc).

Will create a task in Trello for this, for now we can keep this as it is to close this task.

Reply:
BTW, there is a library Querybase, however they claim that Firestore is more featured with querying… 
![image](https://user-images.githubusercontent.com/6065071/47426064-5ed77b80-d7b6-11e8-923c-a5c67f63bbd1.png)

Do you think we may look into Firestore? I didn't want to utilize it because of it's Beta stage and the pricing differences.

Reply:
Firestore looks promising for filtering data how we'd like. I am also concerned with the pricing and Beta stage but do not know the full scale of the project so I can not comment about it.